src/transformers/models/byt5/tokenization_byt5.py

```diff
@@ -196,71 +206,45 @@ def build_inputs_with_special_tokens(
 
     def _tokenize(self, text: str) -> List[str]:
         """Take as input a string and return a list of strings (tokens) for words/sub-words"""
-
-        def _sub_tokenize(sub_text):
-            character_list = list(sub_text)
-            utf_tokens_lists = [list(char.encode("utf-8")) for char in character_list]
-            sub_tokens = [chr(utf_token) for utf_tokens in utf_tokens_lists for utf_token in utf_tokens]
-            return sub_tokens
-
-        # split on special characters
-        pattern = f"({'|'.join(self.special_tokens_encoder.keys())})"
-        sub_texts = list(filter(None, re.split(pattern, text)))
-        tokens = []
-        for sub_text in sub_texts:
-            if sub_text in self.special_tokens_encoder.keys():
-                tokens += [sub_text]
-            else:
-                tokens += _sub_tokenize(sub_text)
-
+        tokens = list(text)
         return tokens
 
     def _convert_token_to_id(self, token):
         """Converts a token (str) in an id using the vocab."""
-        if token.startswith("<extra_id_"):
-            match = re.match(r"<extra_id_(\d+)>", token)
-            num = int(match.group(1))
-            token_id = self.vocab_size - num - 1
-        elif token in self.special_tokens_encoder:
+        if token in self.special_tokens_encoder:
             token_id = self.special_tokens_encoder[token]
-        elif len(token) > 1:
-            # token of length > 1 must be newly added tokens => set them to unk token
+        elif token in self.added_tokens_encoder:
+            token_id = self.added_tokens_encoder[token]
+        elif len(token) != 1:
+            # token of length > 1 must be newly added tokens
```

Comment:
the comments below should probably be removed no?

Reply:
Yes thanks ! Sorry about the cleanup